landd_intergration/landd_integration.py

```diff
@@ -0,0 +1,367 @@
+    enabled Linux environment.
+
+        1) GET secrets from Azure key vault
+        2) GET data from Edx
+        3) Mapping the data
+        4) POST the data to L&D
+
+    """
+
+    def __init__(
+            self,
+            logger=None
+    ):
+
+        # get reference to the user-specified python logger that has already been initialized
+        self.logger = logger
+
+    def log(
+            self,
+            message,
+            message_type="info"
+    ):
+
+        """
+        Log a message
+
+        :param message: the message to log
+        :param message_type: the type of log message (info, warning, error or debug).
+                             If a message is not info or warning, it falls back to debug
+
+        """
+
+        if self.logger:
+            if message_type == "info":
+                self.logger.info(message)
+
+            elif message_type == "warning":
+                self.logger.warning(message)
+            elif message_type == "error":
+                self.logger.error(message)
+            else:
+                self.logger.debug(message)
+
+    def get_access_token(
+            self
+    ):
+
+        """
+
+        Get OAuth2 access token for REST API call using azure MSI extension
+
+        """
+
+        resource = 'https://vault.azure.net'
+        url = 'http://localhost:50342/oauth2/token'
+        data = dict(resource=resource)
+        headers = dict(MetaData='true')
+
+        response = requests.post(url, data=data, headers=headers, timeout=2)
+        if not response.ok:
+            raise RuntimeError(response.content)
+        else:
+            self.log("Got OAuth2 access token using MSI")
+        return response.json()['access_token']
+
+    def get_access_token_ld(
+            self,
+            ldauthorityhosturl,
+            ldtenant,
+            ldresource,
+            ldclientid,
+            ldclientsecret
+    ):
+
+        """
+        Get OAuth2 access token for REST API call for L&D services
+
+        :param ldtenant: tenant id of the AAD application
+        :param ldresource: L&D resource url
+        :param ldclientid: client id of the AAD application
+        :param ldclientsecret: client secret provided by L&D
+        :return: access token
+
+        """
+        authority_url = (ldauthorityhosturl + '/' + ldtenant)
+        context = adal.AuthenticationContext(
+            authority_url, validate_authority=ldtenant != 'adfs',
+            api_version=None)
+
+        token = context.acquire_token_with_client_credentials(
+            ldresource,
+            ldclientid,
+            ldclientsecret)
+        if token['accessToken']:
+            self.log("Got Oauth2 access token for L&D REST API")
+            return token['accessToken']
+        else:
+            raise Exception("Un-handled exception occured while accessing the token for L&D")
+
+    def get_key_vault_secret(
+            self,
+            access_token,
+            key_vault_url,
```

Comment:
i highly recommend
1. making access_token and key_vault_url paramaters of the EdxIntegration constructor and store them as members of the object.
2. removing these params from this function